.github/workflows/frontend_test.yml

```
@@ -0,0 +1,73 @@
+name: Frontend Tests
+
+on:
+  push:
+    paths:
+      - 'frontend/**'
+      - ".github/workflows/frontend*"
+
+  workflow_dispatch:
+
+env:
+  NODE_ENV: production
+  PORT: 3000
+
+jobs:
+  testing-frontend:
+    name: Running frontend tests
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+    defaults:
+      run:
+        working-directory: ${{github.workspace}}/frontend
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'yarn'
+          cache-dependency-path: frontend/yarn.lock
```

Do you think we could read `.nvmrc`?
```suggestion
      - name: Read .nvmrc
        id: nvm
        run: echo "##[set-output name=NVMRC;]$(cat .nvmrc)"
        
      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version: "${{ steps.nvm.outputs.NVMRC }}"
          cache: 'yarn'
          cache-dependency-path: frontend/yarn.lock
```

---

That's a good suggestion, I'll take it